src/basins/attractor_mapping_recurrences.jl

```diff
@@ -143,8 +154,8 @@ end
 #####################################################################################
 # Definition of `BasinInfo` and initialization
 #####################################################################################
-mutable struct BasinsInfo{B, IF, D, T, Q}
-    basins::Array{Int16, B}
+mutable struct BasinsInfo{B, IF, D, T, Q, A <: AbstractArray}
+    basins::A # A always have type parameterization {Int16, B}. Sparse or Array
```

Comment:
```suggestion
mutable struct BasinsInfo{B, IF, D, T, Q, A <: AbstractArray{Int16, B}}
    basins::A # sparse or dense
```
if we can write the same thing with proper code instead of comments, we should prefer the code.